str0m/src/util/value_history.rs

```diff
@@ -0,0 +1,80 @@
+use std::{
+    collections::VecDeque,
+    iter::Sum,
+    ops::AddAssign,
+    time::{Duration, Instant},
+};
+
+/// Holds a history values of type T for up to a certain Duration, as well as the
+/// cumulated (total) value.
+#[derive(Debug)]
+pub(crate) struct ValueHistory<T> {
+    value: T,
+    history: VecDeque<(Instant, T)>,
+    max_time: Duration,
+}
+
+impl<T> ValueHistory<T>
+where
+    T: Copy + AddAssign + Sum,
+{
+    pub fn new(initial: T, max_time: Duration) -> ValueHistory<T> {
+        ValueHistory {
+            value: initial,
+            history: VecDeque::new(),
+            max_time,
+        }
+    }
+
+    /// Adds a timed value
+    /// Note: time should always monotonically increase in subsequent calls to add()
+    pub fn add(&mut self, t: Instant, v: T) {
```

Comment:
Maybe `push` to be consistent with `Vec` and avoiding confusion with "adding numbers together"?

Reply:
sure